lib/ex_force/client/tesla/tesla.ex

```diff
@@ -65,16 +69,20 @@ defmodule ExForce.Client.Tesla do
   """
   @impl ExForce.Client
   def build_oauth_client(instance_url, opts \\ []) do
-    Tesla.client(
-      [
-        {Tesla.Middleware.BaseUrl, instance_url},
-        {Tesla.Middleware.Compression, format: "gzip"},
-        Tesla.Middleware.FormUrlencoded,
-        {Tesla.Middleware.DecodeJson, engine: Jason},
-        {Tesla.Middleware.Headers, get_headers(opts)}
-      ],
-      Keyword.get(opts, :adapter)
-    )
+    custom_middleware = get_from_opts_or_config(opts, :middleware, [])
+
+    adapter = get_from_opts_or_config(opts, :adapter)
+
+    [
+      {Tesla.Middleware.DecodeJson, engine: Jason},
```

Comment:
Do we need to change this order?

Reply:
Changing this order will fix https://github.com/chulkilee/ex_force/issues/70